Add doc files for syncable exports (#1737)
theforeman/foreman-documentation: guides/common/modules/proc_exporting-a-content-view-version-in-syncable-format.adoc

> [id="Exporting_a_Content_View_Version_in_a_Syncable_Format_{context}"]

Reviewer: please ensure the file name matches the title.

= Exporting a Content View Version in a Syncable Format

You can export a version of a Content View to a syncable format that you can use to create your custom CDN and synchronize the content from the custom CDN over HTTP/HTTPS.

ifdef::satellite[]

> You can then serve the generated content via a local webserver on the importing {ProjectServer} or in another {ProjectServer} organization.

Reviewer 2: ```suggestion
You can then serve the generated content through a local webserver on the importing {ProjectServer} or in another {ProjectServer} organization.
```

You cannot directly import Syncable Format exports. Instead, on the importing {ProjectServer} you must:

* Copy the generated content to an HTTP/HTTPS webserver that is accessible to importing {ProjectServer}.
* Update your CDN configuration to *Custom CDN*.
* Set the CDN URL to point to the web server.

> * Optionally set a SSL CA Credential if the web server requires it.

Reviewer 2: ```suggestion
* Optionally set an SSL/TLS CA Credential if the web server requires it.
```
Learned this from Vale. It should be technically correct.

For more information, see xref:Importing_a_Content_View_Version_{context}[].
endif::[]

You can export the following content in a syncable format from {ProjectServer}:

* File repositories
* Kickstart repositories
* RPM repositories

You cannot export Ansible and Docker content.

ifndef::satellite[]
You cannot export DEB repositories.
endif::[]

The export contains the following data:

* Directories containing the RPMs, *listing* files and metadata of each repository in yum format that can be used to synchronize in the importing {ProjectServer}.

.Prerequisites
* Ensure that you set download policy to *Immediate* for all repositories within the Content View you export.
For more information, see xref:Download_Policies_Overview_{context}[].
* Ensure that you synchronize products you export to the required date.
* Ensure that the user exporting the content has the `Content Exporter` role.

.To Export a Content View Version
. List versions of the Content View that are available for export:
+
[subs="+quotes"]
----
# hammer content-view version list \
--content-view="_My_Content_View_" \
--organization="_My_Organization_"

> ---|----------|---------|-------------|-----------------------
> ID | NAME     | VERSION | DESCRIPTION | LIFECYCLE ENVIRONMENTS
> ---|----------|---------|-------------|-----------------------
> 5  | view 3.0 | 3.0     |             | Library
> 4  | view 2.0 | 2.0     |             |
> 3  | view 1.0 | 1.0     |             |
> ---|----------|---------|-------------|----------------------

Reviewer: ```suggestion
```

----

.Export a Content View version
. Get the version number of desired version.
The following example targets version `1.0` for export:
+
[options="nowrap" subs="+quotes"]
----
# hammer content-export complete version \
--content-view="_Content_View_Name_" \
--version=1.0 \

> --organization="_My_Organization_"

Reviewer: ```suggestion
--organization="_My_Organization_" \
```

>  --format=syncable

Reviewer: ```suggestion
--format=syncable
```

----
. Optional: Verify that the exported content is located in the export directory:
+
[options="nowrap" subs="+quotes"]
----

> # ls -lh /var/lib/pulp/exports/_My_Organization_/_Content_View_Name_/1.0/2021-02-25T18-59-26-00-00/

Reviewer: ```suggestion
# ls -lh /var/lib/pulp/exports/_My_Organization_/_My_Content_View_Name_/1.0/2021-02-25T18-59-26-00-00/
```
